orangecontrib/infrared/data.py

```diff
@@ -158,6 +159,23 @@ def read(self):
 
         return _table_from_image(X, features, x_locs, y_locs)
 
+class HDF5Reader_HERMES(FileFormat):
+    """ A very case specific reader for HDF5 files from the HEREMES beamline in SOLEIL"""
+    EXTENSIONS = ('.hdf5',)
+    DESCRIPTION = 'HDF5 file @HERMRES/SOLEIL'
+
+    def read(self):
+        try:
+            hdf5_file = h5py.File(self.filename)
+            energy = np.array(hdf5_file['entry1/Counter0/energy'])
+            intensities = np.array(hdf5_file['entry1/Counter0/data']).T
+            x_locs = np.array(hdf5_file['entry1/Counter0/sample_x'])
```

Comment:
Nice! Nitpicking: I think the try should start over x_locks because that is what is handled in except.